plugin/pkg/admission/initialization/initialization.go

```diff
@@ -43,38 +54,131 @@ type initializerOptions struct {
 	Initializers []string
 }
 
+type InitializationConfig interface {
+	Run(stopCh <-chan struct{})
+	Initializers() (*v1alpha1.InitializerConfiguration, error)
+}
+
 type initializer struct {
-	resources  map[schema.GroupResource]initializerOptions
+	config     InitializationConfig
 	authorizer authorizer.Authorizer
 }
 
-// NewAlwaysAdmit creates a new always admit admission handler
+// Retry config loading failures for up to four and a half seconds if the config hasn't been loaded
+// yet or if the server is down. Creation requests are delayed during this interval, which prevents
+// racy failures during startup until the initializer configuration becomes available.
+// TODO: move into InitializationConfigurationManager, since these values depend on the config
+//   refresh loop.
+const (
+	retryTemporaryConfigFailures = 8
```

Comment:
Why can't we just make the `defaultInterval` and `defaultFailureThreshold` configurable? https://github.com/kubernetes/kubernetes/blob/master/pkg/kubeapiserver/admission/configuration/configuration_manager.go#L29-L30